src/functional/ffront/past_to_itir.py

```diff
@@ -159,50 +214,36 @@ def _visit_stencil_call_out_arg(
                     f"Too many indices for field {out_field_name}: field is {len(node.type.dims)}"
                     f"-dimensional, but {len(out_field_slice_)} were indexed."
                 )
-            domain_args = []
-            for dim_i, (dim, slice_) in enumerate(zip(node.type.dims, out_field_slice_)):
-                # an expression for the size of a dimension
-                dim_size = itir.SymRef(id=_size_arg_from_field(out_field_name.id, dim_i))
-                # lower bound
-                lower = self._visit_slice_bound(
-                    slice_.lower, itir.Literal(value="0", type="int"), dim_size
-                )
-                upper = self._visit_slice_bound(slice_.upper, dim_size, dim_size)
-                if dim.kind == DimensionKind.LOCAL:
-                    raise GTTypeError(f"Dimension {dim.value} must not be local.")
-                domain_args.append(
-                    itir.FunCall(
-                        fun=itir.SymRef(id="named_range"),
-                        args=[itir.AxisLiteral(value=dim.value), lower, upper],
-                    )
-                )
 
+            return (
+                self._construct_itir_out_arg(out_field_name),
+                self._construct_itir_domain_arg(out_field_name, out_field_slice_),
+            )
         elif isinstance(node, past.Name):
-            out_field_name = node
-            domain_args = [
-                itir.FunCall(
-                    fun=itir.SymRef(id="named_range"),
-                    args=[
-                        itir.AxisLiteral(value=dim.value),
-                        itir.Literal(value="0", type="int"),
-                        # here we use the artificial size arguments added to the fencil
-                        itir.SymRef(id=_size_arg_from_field(out_field_name.id, dim_idx)),
-                    ],
+            return (self._construct_itir_out_arg(node), self._construct_itir_domain_arg(node))
+        elif isinstance(node, past.TupleExpr):
+            try:
+                flattened = _flatten_tuple_expr(node)
+            except AssertionError:
```

Comment:
Fine with me, but I wouldn't bother the additional code and just keep the error `_flatten_tuple_expr`. This error surfaces here temporarily only anyway.

Reply:
Implemented your PR with minor changes.